### PR TITLE
CI: update to conan 1.62.0

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

File: .github/workflows/on_PR_linux_matrix.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ jobs:
       - name: install dependencies
         run: |
           sudo apt install -y tree
-          python3 -m pip install conan==1.59.0 ninja
+          python3 -m pip install conan==1.62.0 ninja
```
Comment (kmilos) [inline .github/workflows/on_PR_linux_matrix.yml:29]: Shall we just change all of these to `~=1.62` or `==1.*` and be done with it? This constant extract version babysitting is kinda annoying...

https://packaging.python.org/en/latest/specifications/version-specifiers/#compatible-release

Comment (neheb) [reply .github/workflows/on_PR_linux_matrix.yml]: Done.

**Comment by neheb** [comment]
Failure is unrelated.

**Comment by kmilos** [comment]
I can tell, but never seen it before... Any ideas?

**Review by kmilos — APPROVED** [review]

**Pull request merged**